spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertySourceLocator.java

```diff
@@ -82,14 +83,14 @@ public AwsParamStorePropertySourceLocator(AWSSimpleSystemsManagement ssmClient,
 		String prefix = this.properties.getPrefix();
 
 		String defaultContext = prefix + "/" + this.properties.getDefaultContext();
-		this.contexts.add(defaultContext + "/");
+		String appContext = prefix + "/" + appName;
+
+		addProfiles(this.contexts, appContext, profiles);
```

Comment:
Look first at the comment in the unit test. To match the order we maintain currently I think this should be more like:

```
addProfiles(this.contexts, appContext, profiles);
this.contexts.add(appContext + "/");

addProfiles(this.contexts, defaultContext, profiles);
this.contexts.add(defaultContext + "/");
```

It could be also that I am missing something so please correct me if I am wrong.